pkg/transport/process/extensions/extensions_suite_test.go

```diff
@@ -0,0 +1,161 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+package extensions_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	cdv2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/gardener/component-cli/pkg/transport/process"
+	"github.com/gardener/component-cli/pkg/transport/process/extensions"
+)
+
+const (
+	exampleProcessorBinaryPath = "../../../../tmp/test/bin/example-processor"
+	sleepProcessorBinaryPath   = "../../../../tmp/test/bin/sleep-processor"
+	sleepTimeEnv               = "SLEEP_TIME"
+	sleepTime                  = 5 * time.Second
+)
+
+func TestConfig(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "transport extensions Test Suite")
+}
+
+var _ = BeforeSuite(func() {
+	_, err := os.Stat(exampleProcessorBinaryPath)
+	Expect(err).ToNot(HaveOccurred(), exampleProcessorBinaryPath+" doesn't exists. pls run make install-requirements.")
+
+	_, err = os.Stat(sleepProcessorBinaryPath)
+	Expect(err).ToNot(HaveOccurred(), sleepProcessorBinaryPath+" doesn't exists. pls run make install-requirements.")
+}, 5)
+
+var _ = Describe("transport extensions", func() {
+
+	Context("stdio executable", func() {
+		It("should modify the processed resource correctly", func() {
+			args := []string{}
+			env := []string{}
+			processor, err := extensions.NewStdIOExecutable(exampleProcessorBinaryPath, args, env)
+			Expect(err).ToNot(HaveOccurred())
+
+			runExampleResourceTest(processor)
+		})
+
+		It("should exit with error when timeout is reached", func() {
+			args := []string{}
+			env := []string{
+				fmt.Sprintf("%s=%s", sleepTimeEnv, sleepTime.String()),
```

Comment:
I see that this is the default go syntax that is passed to the command but for the yaml representation wouldn't it be more intuitiv to use a map like
```
env:
  a: value
```
instead of
```
env:
- a=val
```

Reply:
yes correct, i've changed it